src/main/kotlin/com/launchdarkly/intellij/action/CopyKeyAction.kt

```diff
@@ -45,21 +46,27 @@ class CopyKeyAction : AnAction {
      */
     override fun actionPerformed(event: AnActionEvent) {
         val project = event.project
-        var selection = StringSelection("")
         if (project != null) {
-            val selectedNode =
+            var selectedNode =
                 project.service<FlagToolWindow>().getPanel()
-                    .getFlagPanel().tree.lastSelectedPathComponent as DefaultMutableTreeNode
-            if (selectedNode != null) {
-                // Right clicking on Key node. Will break if order changes.
-                if (selectedNode.childCount == 0 && selectedNode.toString().startsWith("Key:")) {
-                    selection = StringSelection(selectedNode.toString().substringAfter(" "))
-                } else if (selectedNode.depth == FLAG_NAME_DEPTH) { //
-                    selection = StringSelection(selectedNode.firstChild.toString().substringAfter(" "))
+                    .getFlagPanel().tree.lastSelectedPathComponent as? DefaultMutableTreeNode
+            while (selectedNode != null) {
+                if (selectedNode.userObject is FlagNodeParent) {
+                    val flagNodeParent = selectedNode.userObject as FlagNodeParent
+                    val selection = StringSelection(flagNodeParent.key)
+                    val clipboard: Clipboard = Toolkit.getDefaultToolkit().systemClipboard
+                    return clipboard.setContents(selection, selection)
+                } else {
+                    selectedNode = selectedNode.parent as? DefaultMutableTreeNode
                 }
-                val clipboard: Clipboard = Toolkit.getDefaultToolkit().systemClipboard
-                clipboard.setContents(selection, selection)
             }
+
+            // If we can't find the key to copy, notify the user
+            val notifier = GeneralNotifier()
+            notifier.notify(
+                project,
+                "Could not copy flag key."
```

Comment:
@launchdarkly/docs-reviewers I have an error string here that I always appreciate your feedback on!

Reply:
LGTM